src/video/x11/SDL_x11messagebox.c

```diff
@@ -233,6 +278,16 @@ static bool X11_MessageBoxInit(SDL_MessageBoxDataX11 *data, const SDL_MessageBox
         }
     }
 
+    if (data->icon_char != '\0') {
+		data->icon_char_font = X11_XLoadQueryFont(data->display, g_IconFont);
+		if (!data->icon_char_font) {
+			data->icon_char_font = X11_XLoadQueryFont(data->display, g_MessageBoxFontLatin1);
+			if (!data->icon_char_font) {
+				return SDL_SetError("Couldn't load icon font %s", g_MessageBoxFontLatin1);
```

Comment:
Instead of failing, we should probably fall back to the existing behavior of not showing an icon.